org.opent2t.sample.hub.superpopular/com.wink.hub/js/thingTranslator.js

```diff
@@ -88,6 +88,152 @@ class Translator {
         // Make the async request
         return this._makeRequest(requestPath, 'PUT', putPayloadString);
     }
+    
+    /**
+     * Subscribes to a Wink pubsubhubbub feed.  This function is designed to be called twice.
+     * The first call will contain just the callbackUrl, which will be subscribed for postbacks
+     * from the Wink cloud service.
+     * The second call will contain the request, and response for a GET to the callbackUrl that was
+     * provided in order to complete the verification of the subscription.
+     * 
+     * The expected sequence is as follows:
+     * - Call subscribe with a callback url, get a response.
+     * - Http server running at the callback url receives a GET request, this request and a response string
+     *      are passed to a 2nd call to subscribe() (callbackUrl is ignored in this case)
+     * - Http server responds to the GET with the verificationResponseContent that was populated by subscribe
+     * - Future POST calls to the callback url are passed to the device translator get*Uri function
+     * 
+     * All calls to subscribe will return the expiration time of the subscription.  For Wink, this is
+     * currently 24 hours.  Subsequent calls to subscribe will refresh the expiration time, and in the
+     * case of Wink/Pubsubhubbub will not require verification.
+     * 
+     * @param {string} deviceType - Device Type (e.g. 'thermostats')
+     * @param {string|number} deviceId - Id for the specific device
+     * @param {string} callbackUrl - Callback url for feed postbacks
+     * @param {HttpRequest} verificationRequest - GET request received by the server at a previously subscribed
```

Comment:
The entire request.  For this specific instance, Wink only needs parameters from the GET request, but this may not be true for other protocols (none of which we are using yet).  We want the callback server to do as little work as possible, so we'll pull the parts we need from the request as needed.

---
In reply to: [80521932](https://github.com/openT2T/translators/pull/91#discussion_r80521932) [](ancestors = 80521932)